core/src/mock_alpenglow_consensus.rs

```diff
@@ -586,7 +585,7 @@ impl MockAlpenglowConsensus {
                         let mut lockguard = get_state_for_slot(&state, slot).lock().unwrap();
                         // check if tasks have been aborted and do not report garbage
                         if lockguard.current_slot == 0 {
-                            return;
+                            continue;
```

Comment:
can you explain how these changes fix the problem?

Reply:
where does the state get cleaned up? I thought that was happening as part of `lockguard.reset();`

Reply:
Yes, correct. The runner thread does it. The issue was that it never got the command to run -> never cleaned.

Reply:
lockguard.current_slot == 0 implies the slot is not initialized at all (since any legit slot has non-zero index).  

> What happens if we don't and do the continue like you have here? Where does the state eventually get cleaned up?
In this case, if lockguard.current_slot == 0 it would imply that the state does not need to be cleared at all. And now I clearly see why this is confusing as hell, I'll refactor.